akka-actor/src/main/scala-2.13+/akka/util/ByteString.scala

```diff
@@ -386,6 +388,15 @@ object ByteString {
       }
     }
 
+    override def copyToArray[B >: Byte](xs: Array[B], start: Int, len: Int): Int = {
+      // min of the bytes available top copy, bytes there is room for in dest and the requested number of bytes
+      val bytesToCopy = length.min(len).min(xs.length - start)
+      if (bytesToCopy > 0) {
+        Array.copy(bytes, startIndex, xs, start, bytesToCopy)
```

Comment:
would `java.lang.System#arraycopy` be even faster or is that not safe here?

Reply:
Array.copy is using `System#arraycopy` after some checks. Here we know its bytes so maybe we could skip the checks and use `arrayCopy` directly. Or is it something with the `B >: Byte`?

Reply:
I'll try using that directly. Not quite sure why it is a type bound rather than just Byte.

Reply:
All tests passes with `System#arrayCopy` but I guess that if someone calls it with an array of boxed Bytes it will blow up somehow?